processor/resourcedetectionprocessor/internal/resourcedetection.go

```diff
@@ -116,6 +124,14 @@ func (p *ResourceProvider) Get(ctx context.Context, client *http.Client) (resour
 	return p.detectedResource.resource, p.detectedResource.schemaURL, p.detectedResource.err
 }
 
+func (p *ResourceProvider) ConfigureHandlers(ctx context.Context, host component.Host, middlewareId component.ID) {
+	for _, detector := range p.detectors {
+		if handlerDetector, ok := detector.(ExposeHandlerDetector); ok {
+			awsmiddleware.TryConfigure(p.logger, host, middlewareId, awsmiddleware.SDKv1(handlerDetector.ExposeHandlers(ctx)))
```

Comment:
nit: Could grab the `Configurer` once and pass that into this function instead of the host and middlewareID so it isn't trying to get it each time.